airflow/www/views.py

```diff
@@ -1244,7 +1250,12 @@ def rendered_templates(self, session):
         task_id = request.args.get('task_id')
         map_index = request.args.get('map_index', -1, type=int)
         execution_date = request.args.get('execution_date')
-        dttm = timezone.parse(execution_date)
+
+        try:
+            dttm = timezone.parse(execution_date)
+        except ParserError:
+            return Response("Invalid execution_date", mimetype="text/plain", status=400)
```

Comment:
Actually, maybe we can make use of `werkzeug.exceptions.BadRequest`:

```python
def _safe_parse_datetime(v: str) -> datetime.datetime:
    try:
        return timezone.parse(v)
    except ParserError:
        return BadRequest(f"Invalid datetime: {v!r}")
```

And always use this instead of `timezone.parse()` directly. Flask catches this exception and return 400 automatically for us.

Reply:
Thanks for the tip, that will reduce redundant parsing code spread across all functions and keep logic at one place.